src/zedbit/mailx/Mailx.java

```diff
@@ -0,0 +1,506 @@
+        traverse (page, pageUrl);
+
+    } // End crawl
+
+    /**
+    * <p>traverse checks if the current page has been visited before (by checking against 
+    * the contents of the <code>urlsVisited</code> set). If the page has not been visited,
+    * traverse looks for text and comment nodes that might
+    * contain email strings by calling <code>searchNodes<code> on all HTML nodes containing 
+    * text, comments included.</p>
+    * 
+    * <p>In addition, static links are collected, and dynamic routes visited via simulating 
+    * HtmlUnit's click action. <code>traverse</code> then calls <code>visitDynamicLinks</code> 
+    * and <code>visitStaticLinks</code>on these reachable links, which results in recursion.</p>
+    */
+    private static void traverse (HtmlPage page, String pageUrl) {
+
+        // Mark we have been here!
+        urlsVisited.add(pageUrl);
+
+        // The page is open and loaded, now we can search...
+        searchNodes(page, pageUrl, "//text()[contains(.,\"@\")]"); // all DOM text nodes &
+        searchNodes(page, pageUrl, "//comment()[contains(.,\"@\")]"); // all html comments
+
+        try {
+
+            // Now get all static links to crawl along these, accumulate then in urlsReachable
+            getStaticLinks (page, pageUrl);
+
+            // Here we click and crawl recursively on all dynamic links
+            visitDynamicLinks (page, pageUrl);
+    
+            // Now crawl recursively on all static links
+            visitStaticLinks (page, pageUrl);
+
+        } catch (StackOverflowError e) {
+            printlnV("StackOverflow to get to this link depth! Increase, if possible -Xss param at the command line!");
+            if (trace) { e.printStackTrace(); }
+            return;
+        } catch (Exception e) {
+            printlnV("Unexpected error while visiting and traversing links!");
+            if (trace) { e.printStackTrace(); }
+            return;
+
+        }
+
+    }
+
+    /**
+    * searchNodes visits the current page for DOM nodes, and then looks 
+    * looking for strings that might be emails in each of these nodes.mic routes
+    */
+    private static void searchNodes (HtmlPage page, String pageUrl, String xPath) {
+
+
+        List<?> possibles;
+        try { possibles = page.getByXPath(xPath); }
+        catch (Exception e) {
+            printlnV("HtmlUnit's searchNode XPATH=" + xPath + " died for this particular page! ");
+            if (trace) { e.printStackTrace(); }
+            return;
+        }
+        printlnT("---- Possibilities found: " + possibles.size());
+
+        Set<String> emails = new HashSet<String>();
+        possibles.forEach((i) -> {
+            String iStr = i.toString();
+            printlnT(MARGIN + iStr);
+            Matcher m = emailP.matcher(iStr); 
+
+            // Peel the first, second, third, or so email present in this text node
+            while (m.find() == true) { 
```

Comment:
Issue addressed in commit 5a6ca3d

I was trying to peel lines that contained more than one email, and was trying to do so with find() and regex groups. I think it is much cleared with split and made a change. Thus when the code finds something like <!-- a@b.com, c@d.com --> the split works.

The code is now a bit faster without the additional regex, and solves problems where emails have embedded dashes/minus chars, such as arctan-info@mit.edu